php/Exchange.php

```diff
@@ -2521,4 +2521,12 @@ function base32_decode($s){
         $otp = $code % pow(10, 6);
         return str_pad((string) $otp, 6, '0', STR_PAD_LEFT);
     }
+
+    public static function setDefault($array, $key, $default_value) {
```

Comment:
There's one issue with this method in PHP that makes it different from other languages – in PHP arrays aren't copied by reference, but by value instead, so, when you pass the object into the method in JS – it's the reference to the same object, however, when you pass it into the PHP function – it's a copy (a different object from there). The same thing happens upon the exit from this codeblock, so the returned array is copied in PHP. This type of inconsistency between implementations of the same method in different languages can lead to certain hard-to-catch bugs further down the road. We should either make it accept and return a reference, or we should copy the object everywhere in all languages.

Also, I'd suggest to use uniform naming across languages – I know there are other inconsistencies with naming out there, but we're not proud of those and should fix them as well.

Lmk your thoughts.

Reply:
i dont think it is possible to return references to arrays in php, going to close this...

Reply:
But it's actually possible with the `&` symbol: https://www.php.net/manual/en/language.references.return.php. Feel free to reopen this if you want.

Reply:
I tried but it seems that whenever you assign something to array the array gets copied. I think this will only cause more confusion ::")